test/functional/users_controller_test.rb

```diff
@@ -364,6 +365,39 @@ class UsersControllerTest < ActionController::TestCase
     assert_response :unprocessable_entity
   end
 
+  context "an active session is present" do
+    test "#login redirects to root(dashboard)" do
+      get :login, set_session_user
+
+      get :login
+      assert_redirected_to root_path
+    end
+
+    test "for user :one -> #login redirects to root(dashboard)" do
+      get :login, set_session_user.merge(user: users(:one).id)
+
+      get :login
+      assert_redirected_to root_path
+    end
+  end
+
+  context "no active session present" do
+    test "#login renders login page" do
```

Comment:
line #324 tests the response of :login
In above case the test is on rendered template. Maybe I can extend #324 case itself. Wont't that violate 1 assert per test standard ?

```ruby
  test "#login renders login page" do                                                                                                                                                         
    session.clear                                                                                                                                                                             
    get :login                                                                                                                                                                                
    assert_response :success                                                                                                                                                                  
  end
```